engine/execution/provider/engine.go

```diff
@@ -168,7 +158,26 @@ func (e *Engine) onChunkDataRequest(
 	}
 
 	if err != nil {
-		return fmt.Errorf("could not retrieve chunk ID (%s): %w", origin, err)
+		return fmt.Errorf("could not retrieve chunk ID (%s): %w", originID, err)
+	}
+
+	blockID, err := e.execState.GetBlockIDByChunkID(cdp.ChunkID)
+	if err != nil {
+		return fmt.Errorf("cannot find blockID corresponding to chunk data pack ID (%s): %w", cdp.ID(), err)
+	}
+
+	origin, err := e.state.AtBlockID(blockID).Identity(originID)
+	if err != nil {
+		return engine.NewInvalidInputErrorf("invalid origin id (%s): %w", origin, err)
+	}
+
+	// only verifier nodes are allowed to request chunk data packs
+	if origin.Role != flow.RoleVerification {
+		return engine.NewInvalidInputErrorf("invalid role for receiving collection: %s", origin.Role)
+	}
+
+	if origin.Stake == 0 {
+		return engine.NewInvalidInputErrorf("node %s is not staked for the epoch corresponding to the requested chunk data pack", origin.NodeID)
```

Comment:
This part can be encapsulated in a method [like this](https://github.com/onflow/flow-go/pull/427/files#diff-e2a1bac627623a345dc4e9caf84fa096cdecddbe535730904d7438062ffb3820R246-R259). The advantage is that the returned error can be logged at the invoker (i.e., `onChunkDataRequest`) accompanied with more details about the chunk data pack request.